src/lightning_utilities/core/apply_func.py

```diff
@@ -20,6 +20,14 @@ def is_dataclass_instance(obj: object) -> bool:
     return dataclasses.is_dataclass(obj) and not isinstance(obj, type)
 
 
+def can_roundtrip_sequence(obj: Sequence) -> bool:
+    """Check if sequence can be roundtripped."""
+    try:
+        return obj == type(obj)(list(obj))  # type: ignore[call-arg]
```

Comment:
How can you be sure that `list(obj)` will work for any sequence? Even though this is covered by the try-except block, the `and` conditions added to L129 will now skip the is_sequence loop for any sequence that this doesn't consider

Given the complexity and that your data structure is very specific to your problem. Could we instead expose an API so that you can have your data structure define whether it should be treated as a sequence or not?

For instance, that is similar to how pytrees are implemented which are much more extensible and performant: https://github.com/pytorch/pytorch/blob/main/torch/utils/_pytree.py#L163. In this case, it could also be an attribute of the data structure

Reply:
> Given the complexity and that your data structure is very specific to your problem. Could we instead expose an API so that you can have your data structure define whether it should be treated as a sequence or not?

That sounds like the best solution. An exclude list for types that should just be passed through would be ideal. I can work on a PR in early June.

Reply:
so how is it going here? :)

Reply:
It's still on the list. I'm just lacking time to work on open-source these days...

I'll close this PR and will open a new one with exlude support when I find the time.